authz/sdk_end2end_test.go

```diff
@@ -418,23 +417,34 @@ func (s) TestSDKFileWatcher_ValidPolicyRefresh(t *testing.T) {
 
 	// Rewrite the file with a different valid authorization policy.
 	valid2 := sdkTests["AllowsRpcEmptyDenyMatchInAllow"]
-	if err := ioutil.WriteFile(file, []byte(valid2.authzPolicy), os.ModePerm); err != nil {
-		t.Fatalf("ioutil.WriteFile(%q) failed: %v", file, err)
+	if err := os.WriteFile(file, []byte(valid2.authzPolicy), os.ModePerm); err != nil {
+		t.Fatalf("os.WriteFile(%q) failed: %v", file, err)
 	}
-	// Wait 2 seconds for background go routine to read the updated files.
-	time.Sleep(2 * time.Second)
+	// Wait 30 ms for background go routine to read the updated files.
+	time.Sleep(30 * time.Millisecond)
```

Comment:
You shouldn't need an extra delay here since you are looping to retry and have a delay there.

Reply:
Removed the extra delay. So currently in the loop, I have added delay of 100 ms. If we want to reduce that to 10ms say, to have more frequent retries, then we would need this wait time for 100 or 100+ ms since the reload interval is 100 ms.

Reply:
> If we want to reduce that to 10ms say, to have more frequent retries, then we would need this wait time for 100 or 100+ ms since the reload interval is 100 ms.

I don't think that's quite right.  The reload interval could be very close to elapsing naturally already.  I think it's fine to delay 10-50ms in the retry loop with a 100ms reload frequency.  The first iteration doesn't need to succeed; that's why we retry.  A shorter delay on retries will ensure we pass the test more closely to the moment the reload has taken effect.

Reply:
I agree, shorter retry helps pass the test closer to when reload takes place. My earlier concern was that, with refresh duration 100 ms, if we directly start the retries(say 20 ms) without any wait, initial say 5 retries may be redundant. But yeah, with such small intervals, this is not a concern.